src/Appwrite/Platform/Tasks/Restore.php

```diff
@@ -0,0 +1,300 @@
+
+        $stdout = '';
+        $stderr = '';
+        $cmd = 'tar -xzf ' . $file . ' -C ' . $directory;
+        Console::execute($cmd, '', $stdout, $stderr);
+        if (!empty($stderr)) {
+            Console::error($stderr);
+            Console::exit();
+        }
+
+        if (!file_exists($file)) {
+            Console::error('Restore file not found: ' . $file);
+            Console::exit();
+        }
+    }
+
+    public function decompress(string $target)
+    {
+        $this->log('Decompress start');
+
+        $logfile = $target . '/../log.txt';
+
+        $args = [
+            'xtrabackup',
+            '--decompress',
+            '--strict',
+            '--remove-original',
+            '--compress-threads=' . $this->processors / 2,
+            '--parallel=' . $this->processors,
+            '--target-dir=' . $target,
+            '2> ' . $logfile,
+        ];
+
+        $cmd = 'docker exec ' . $this->xtrabackupContainerId . ' ' . implode(' ', $args);
+        shell_exec($cmd);
+
+        $stderr = shell_exec('tail -1 ' . $logfile);
+
+        if (!str_contains($stderr, 'completed OK!')) {
+            Console::error('Decompress failed');
+            Console::exit();
+        }
+    }
+
+    public function prepare(string $target)
+    {
+        $this->log('Prepare start');
+
+        if (!file_exists($target)) {
+            Console::error('prepare error directory not found: ' . $target);
+            Console::exit();
+        }
+
+        $logfile = $target . '/../log.txt';
+
+        $args = [
+            'xtrabackup',
+            '--prepare',
+            '--parallel=' . $this->processors,
+            '--strict',
+            '--target-dir=' . $target,
+            '2> ' . $logfile,
+        ];
+
+        $cmd = 'docker exec ' . $this->xtrabackupContainerId . ' ' . implode(' ', $args);
+        shell_exec($cmd);
+
+        $stderr = shell_exec('tail -1 ' . $logfile);
+
+        if (!str_contains($stderr, 'completed OK!')) {
+            Console::error(date('Y-m-d H:i:s') . ' Prepare failed:' . $stderr);
+            Console::exit();
+        }
+    }
+
+    public function restore(string $target, bool $cloud, string $datadir)
+    {
+        $this->log('Restore start');
+
+        if (!file_exists($target)) {
+            Console::error('restore error directory not found: ' . $target);
+            Console::exit();
+        }
+
+        $logfile = $target . '/../log.txt';
+
+        $args = [
+            'xtrabackup',
+            $cloud ? '--move-back' : '--copy-back',
+            '--strict',
+            '--parallel=' . $this->processors,
+            '--target-dir=' . $target,
+            '--datadir=' . $datadir,
+            '2> ' . $logfile,
+        ];
+
+        $cmd = 'docker exec ' . $this->xtrabackupContainerId . ' ' . implode(' ', $args);
+        shell_exec($cmd);
+
+        $stderr = shell_exec('tail -1 ' . $logfile);
+
+        if (!str_contains($stderr, 'completed OK!')) {
```

Comment:
Let's add a retry loop with 5 max attempts here

Reply:
If move-back fails in the middle, that means Datadir will have content inside and retiring will always fail , since it must be empty